Assignment6.Rmd

```diff
@@ -1,86 +1,180 @@
+
+To terminate the new tree at two nodes, I chose cp = 0.9, as the next node will only improve cp by 0.923 - 0.058 = 0.865 < 0.9. (Recall: cp means any split that does not decrease the overall lack of fit by a factor of cp is not attempted. The user informs the program that any split which does not improve the fit by cp will likely be pruned off by cross-validation, and that hence the program need not pursue it.)
 
 ```{r}
-M2$predict1 <- predict(c.tree1, M2, type = "class")
+rpart2 <- prune(rpart1, cp = 0.9)#Set cp to the level at which you want the tree to end
+
+printcp(rpart1)
 
-M2$predict2 <- predict(c.tree2, M2, type = "class")
+rpart.plot(rpart2)
+post(rpart2, file = "tree2.ps", title = "MOOC") #This creates a pdf image of the tree
+```
+
+####Now use both the original tree and the pruned tree to make predictions about the the students in the second data set. Which tree has a lower error rate?
 
+```{r}
+M2$predict1 <- predict(rpart1, M2, type = "class")
 table(M2$certified, M2$predict1)
 
+M2$predict2 <- predict(rpart2, M2, type = "class")
 table(M2$certified, M2$predict2)
 
 ```
 
+rpart1:
+
+Error rate (ERR) = (FP+FN)/(P+N) = (7790+24)/(2056+24+7790+130) = 0.7814
+Accuracy (ACC) = 1-ERR = 0.2186
+
+rpart2:
+
+Error rate (ERR) = (FP+FN)/(P+N) = (3453+1184)/(896+1184+3453+4467) = 0.4637
+Accuracy (ACC) = 1-ERR = 0.5363
+
+rpart2 has a lower error rate. This is expected, as the pruning of nodes reduces overfitting and increases the ability to predict.
+
 ##Part III
 
-Choose a data file from the (University of Michigan Open Data Set)[https://github.com/bkoester/PLA/tree/master/data]. Choose an outcome variable that you would like to predict. Build two models that predict that outcome from the other variables. The first model should use raw variables, the second should feature select or feature extract variables from the data. Which model is better according to the cross validation metrics?
+####Choose a data file from the (University of Michigan Open Data Set)[https://github.com/bkoester/PLA/tree/master/data]. Choose an outcome variable that you would like to predict. Build two models that predict that outcome from the other variables. The first model should use raw variables, the second should feature select or feature extract variables from the data. Which model is better according to the cross validation metrics?
 
-```{r}
+I selected variables that I wanted to examine across the two data files, and combined it into a data set called "df". It initially included `GPAO`, but it was the only variable used in the tree construction. Hence, I removed it for the purpose of this exercise. 
+
+(Question: what does it mean when there are multiple variables, but only 1 is used in tree construction? Understand it means cp does not reduce lack of fit by 0.01 with the next split, but does that mean it's the most important variable?)
```

Comment:
According to one source (see the link below), the "`rpart` algorithm works by splitting the dataset recursively, which means that the subsets that arise from a split are further split until a predetermined termination criterion is reached. At each step, the split is made based on the independent variable that results in the largest possible reduction in heterogeneity of the dependent (predicted) variable." In other words, it is a variable that results in the most pure nodes. Variables that don't appear in the tree are those that don’t give any new subsets that would be interesting for that tree.  
In regards to the variable importance, "An overall measure of variable importance is the sum of the goodness of split measures for each split for which it was the primary variable, plus goodness * (adjusted
agreement) for all splits in which it was a surrogate." You can use `varImp()` function from the `caret` package to calculate the variable importance. 
Link: https://eight2late.wordpress.com/2016/02/16/a-gentle-introduction-to-decision-trees-using-r/
Link: https://cran.r-project.org/web/packages/rpart/vignettes/longintro.pdf

Hopefully that helps.